### PR TITLE
Physical_Engine: Add support for GeneralMaterialTakeoff

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/1516   

### Issues addressed by this PR

- Adds first support methods and collection system for GeneralMaterialTakeoff class
- Add Aggregation and mapping support for GeneralMaterialTakeoff
- Add GeneralTakeoff capability for physical surface obejcts
- Add method to generate a general takeoff given a construction and an area.

### Test files

https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/BHoM/Physical_oM/%231516-AddGeneralTakeoffObject?csf=1&web=1&e=iAsS7e

### CODE REVIEW

File: Matter_Engine/Query/GeneralMaterialTakeoff.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                VolumetricMaterialTakeoff volMatTakeoff;
+                if (TryGetVolumetricMaterialTakeoffFragment(elementM, out volMatTakeoff))
+                    return Create.GeneralMaterialTakeoff(volMatTakeoff);
+            }
+            //IElementMs should implement one of the following:
+            // -SolidVolume and MaterialComposition or
+            // -VolumetricMaterialTakeoff
+            //- GeneralMaterialTakeoff
+            //This method first checks if the VolumetricMaterialTakeoff method can be found and run, and if so uses it.
+            //If not, it falls back to running the MaterialComposition and SolidVolume methods and gets the VolumetricMaterialTakeoff from them.
```
Comment (michaelhoehn) [inline Matter_Engine/Query/GeneralMaterialTakeoff.cs:67]: @IsakNaslundBh worth adding this to the documentation for required extension methods post-merge

**Review by michaelhoehn — DISMISSED** [review]
I've tested this with the provided sample script and everything works as expected. Nicely done @IsakNaslundBh 👍

File: Matter_Engine/Query/GeneralMaterialTakeoff.cs (hunk 1 above)
Comment (IsakNaslundBh) [reply Matter_Engine/Query/GeneralMaterialTakeoff.cs:67]: Yes, was planning to, but good reminder. Thanks :)

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance
@BHoMBot check unit-tests
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot check unit-tests
@BHoMBot check code-compliance
@BHoMBot check copyright-compliance
@BHoMBot check documentation-compliance
@BHoMBot check project-compliance
@BHoMBot check core
@BHoMBot check null-handling
@BHoMBot check serialisation

**Review by alelom — APPROVED** [review]
Approving https://github.com/BHoM/BHoM_Engine/pull/3047/commits/615ffc42a6d8d1c67beedb582ee310c47eead1ae and https://github.com/BHoM/BHoM_Engine/pull/3047/commits/675fe72ba6aed8c93b150ad7602afa04f94846a6 based on code changes and the previous approving review.

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: dataset-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**